src/background/config_backend.js

```diff
@@ -0,0 +1,87 @@
+import Logger from "../logger.js";
+import DBUtils from "../utilities.js";
+
+const defaultConfig = {
+  displayWaveform: false
+};
+
+export default class ConfigBackend {
+  constructor() {
+    this.log = new Logger();
+    this.dbUtils = new DBUtils();
+
+    this.boundConnectionListenerCallback = ConfigBackend.connectionListenerCallback.bind(
+      this
+    );
+
+    this.boundPortListenerCallback = ConfigBackend.portListenerCallback.bind(
+      this
+    );
+  }
+
+  init() {
+    this.log.info("initializing ConfigBackend");
+
+    chrome.runtime.onConnect.addListener(this.boundConnectionListenerCallback);
+  }
+
+  static connectionListenerCallback(port) {
+    this.log.info("connection listener callback");
+    if (port.name !== "bandcamplabelview") {
+      this.log.error(
```

Comment:
What's the purpose of this error? Can `port.name` ever _not_ equal `bandcamplabelview`?

Reply:
@emcniece yeah, as far as I know this API is not sandboxed so the callback can be triggered by other plugins that are also using this communication system.

Reply:
Ah, interesting. So we don't necessarily care that the port is the wrong name (hence not throwing here), we care if a port named `bandcamplabelview` is never opened. That's a bit harder to test for I think.